prowler/providers/aws/services/efs/efs_not_publicly_accessible/efs_not_publicly_accessible.py

```diff
@@ -19,23 +19,48 @@ def execute(self):
                 report.status = "FAIL"
                 report.status_extended = f"EFS {fs.id} doesn't have any policy which means it grants full access to any client."
             else:
-                for statement in fs.policy["Statement"]:
-                    if statement["Effect"] == "Allow":
-                        if (
-                            ("Principal" in statement and statement["Principal"] == "*")
-                            or (
-                                "Principal" in statement
-                                and "AWS" in statement["Principal"]
-                                and statement["Principal"]["AWS"] == "*"
-                            )
-                            or (
-                                "CanonicalUser" in statement["Principal"]
-                                and statement["Principal"]["CanonicalUser"] == "*"
-                            )
-                        ):
-                            report.status = "FAIL"
-                            report.status_extended = f"EFS {fs.id} has a policy which allows access to everyone."
-                            break
+                for statement in fs.policy.get("Statement", []):
+                    if statement.get(
+                        "Effect"
+                    ) == "Allow" and self.is_public_access_allowed(statement):
+                        report.status = "FAIL"
+                        report.status_extended = (
+                            f"EFS {fs.id} has a policy which allows access to everyone."
+                        )
+                        break
             findings.append(report)
-
         return findings
+
+    def is_public_access_allowed(self, statement):
+        principal = statement.get("Principal")
+        if principal == "*" or (
+            isinstance(principal, dict) and "*" in principal.values()
+        ):
+            return not self.has_secure_conditions(statement)
+        return False
+
+    def has_secure_conditions(self, statement):
+        conditions = statement.get("Condition", {})
+        allowed_conditions = {
+            "aws:SourceArn",
+            "aws:SourceVpc",
+            "aws:SourceVpce",
+            "aws:SourceOwner",
+            "aws:SourceAccount",
+        }
+        if (
+            "Bool" in conditions
+            and conditions["Bool"].get("elasticfilesystem:AccessedViaMountTarget")
+            == "true"
+        ):
+            return True
```

Comment:
It'd be more readable to set a `has_secure_conditions = False` by default and the here
```suggestion
            has_secure_conditions = True
```